File: Longest_SubArray.js
//Find the maximum length of the a subarray/window

function Find_Length(arr, target) {
    currSum = 0;
    ansArr = 0;
    left = 0;
    
    for (let right = 0; right < arr.length; right++) {
        currSum += arr[right];

        while (currSum > target) {
            currSum -= arr[left];
            left += 1;
        }
        ansArr = Math.max(ansArr, right - left + 1)
    }
    return ansArr;
}

console.log(Find_Length([3, 1, 2, 7, 4, 2, 1, 1, 5], 8));

File: LeetCode/Longest_SubArray.py
'''
Longest_SubArray.pyGiven an array of positive integers nums and an integer k, 
find the length of the longest subarray whose sum is less than or equal to k. 
This is the problem we have been talking about above. We will now formally solve it.
'''
from typing import List

def find_length(nums: List[int], k: int) -> int:
    currSum = 0
    left = 0
    ansArray = 0
    
    for right in range(len(nums)):
        currSum += nums[right]
        
        while currSum > k:
            currSum -= nums[left]
            left += 1
        ansArray = max(ansArray, (right - left + 1))
    return ansArray

arr = [3, 1, 2, 7, 4, 2, 1, 1, 5]
k = 8.
print(find_length(arr, k))
      